docs/user-guide/persistent-volumes/index.md

```diff
@@ -216,3 +290,116 @@ spec:
 ### A Note on Namespaces
 
 `PersistentVolumes` binds are exclusive, and since `PersistentVolumeClaims` are namespaced objects, mounting claims with "Many" modes (`ROX`, `RWX`) is only possible within one namespace.
+
+## StorageClasses
+
+Each `StorageClass` contains the fields `provisioner` and `parameters`, which
+are used when a `PersistentVolume` belonging to the class needs to be
+dynamically provisioned.
+
+The name of a `StorageClass` object is significant, and is how users can
+request a particular class. Administrators set the name and other parameters
+of a class, all of which are opaque to users, when first creating
+`StorageClass` objects, and the objects cannot be updated once they are
+created.
+
+Administrators can specify a default `StorageClass` just for PVCs that don't
+request any particular class to bind to: see the
+[`PersistentVolumeClaim` section](docs/user-guide/persistent-volumes/#class-1)
+for details.
+
+```yaml
+kind: StorageClass
+apiVersion: extensions/v1beta1
+metadata:
+  name: standard
+provisioner: kubernetes.io/aws-ebs
+parameters:
+  type: gp2
+```
+
+### Provisioner
+Storage classes have a provisioner that determines what volume plugin is used
+for provisioning PVs. This field must be specified. During beta, the available
+provisioner types are `kubernetes.io/aws-ebs` and `kubernetes.io/gce-pd`.
+
+### Parameters
+Storage classes have parameters that describe volumes belonging to the storage
+class. Different parameters may be accepted depending on the `provisioner`. For
+ example, the value `io1`, for the parameter `type`, and the parameter
+`iopsPerGB` are specific to EBS. When a parameter is omitted, some default is
+used.
+
+#### AWS
+
+```yaml
+kind: StorageClass
+apiVersion: extensions/v1beta1
+metadata:
+  name: slow
+provisioner: kubernetes.io/aws-ebs
+parameters:
+  type: io1
+  zone: us-east-1d
+  iopsPerGB: "10"
+```
+
+* `type`: `io1`, `gp2`, `sc1`, `st1`. See AWS docs for details. Default: `gp2`.
```

Comment:
we need a huge link to AWS docs (http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html) is necessary, rules about available sizes and types (and IOPS) are quite complicated.